### DISCUSSION TITLE
HA k3s cluster launch with external DB via terraform

### DISCUSSION THREAD

**Discussion by huckym** [k3s-io/k3s#6187] [General]
If I start a cluster with terraform with a for_each/count statement to launch multiple servers, would I have to worry about parallelism? or does k3s manage concurrency synchronization of DB access during cluster creation?

It would be a shame to work-around on terraform side with -parallelism=1

**Comment by brandond** [comment]
No, there is no locking around initialization of the database. If you put a random wait of 0-5 seconds on each of them, or staggered them by a few seconds, that should be fine. You just don't want them all trying to do initial database setup at the exact same time.
